### PR TITLE
add tidb versioning doc

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Add the doc for TiDB versioning system. 

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [x] v6.0 (TiDB 6.0 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/9238 https://github.com/pingcap/docs-cn/pull/9361
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: releases/versioning.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+title: TiDB Versioning
+summary: Learn the version numbering system of TiDB.
+---
+
+# TiDB Versioning
+
+<Important>
+It is recommended to always upgrade to the latest patch release of your release series.
+</Important>
```
Comment (ran-huang) [inline releases/versioning.md:10]: This is to avoid rendering failure.

File: releases/versioning.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Example versions:
+
+- 1.1 Beta
+- 2.1 Beta
+- 4.0.0-beta.1
+
+### Alpha Releases
+
+Alpha Releases are internal releases for testing and introduce new features and improvements.Alpha releases are the initial versions of the current release series. Alpha Releases might have some bugs and are available for users to test the latest features.
```
Comment (ran-huang) [inline releases/versioning.md:71]:
```suggestion
Alpha Releases are internal releases for testing and introduce new features and improvements. Alpha releases are the initial versions of the current release series. Alpha Releases might have some bugs and are available for users to test the latest features.
```

File: releases/versioning.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+<Important>
+It is recommended to always upgrade to the latest patch release of your release series.
+</Important>
+
+Starting from TiDB v6.0.0, TiDB has introduced a new release series, DMR. Currently TiDB has two release series:
+
+* Long-Term Support Releases (LTS)
+* Development Milestone Releases (DMR)
+
+For the versioning system of TiDB v5.0.0 and earlier versions, refer to [Historical versioning](#historical-versioning-deprecated).
```
Comment (lilin90) [inline releases/versioning.md:17]: This description doesn't cover v5.1-5.4 & 5.0.x. It would be better to make this clear.

**Comment by ran-huang** [comment]
@lilin90 @qiancai I have incorporated changes from https://github.com/pingcap/docs-cn/pull/9361. PTAL, thanks!

File: releases/versioning.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+Example versions:
+
+- 6.1.0
+- 5.4.0
+
+During the lifecycle of LTS, patch releases are made available on demand. Patch releases contain bug fixes and security vulnerability fixes, and do not introduce new features.
+
+Patch releases are versioned as `X.Y.Z`. `X.Y` is consistent with the corresponding LTS versioning. The patch number `Z` increments from 1.
+
+Example versions:
```
Comment (lilin90) [inline releases/versioning.md:42]:
```suggestion
Example version:
```

File: releases/versioning.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+v5.1.0, v5.2.0, v5.3.0, v5.4.0 were released only two months after their preceding releases, but all four releases are LTS and provide patch releases.
+</Note>
+
+## Development Milestone Releases
+
+Development Milestone Releases (DMR) are released approximately every two months that do not contain LTS. DMR introduce new features, improvements and bug fixes. TiDB does not provide patch releases based on DMR, and any related bugs are fixed in the subsequent release series.
+
+DMR are versioned as `X.Y.Z`. `Z` defaults to 0. A `-DMR` suffix is appended to the version number.
+
+Example versions:
```
Comment (lilin90) [inline releases/versioning.md:56]:
```suggestion
Example version:
```

File: releases/versioning.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+- 6.0.0-DMR
+
+## Versioning of TiDB ecosystem tools
+
+Some TiDB tools are released together with TiDB server and use the same version numbering system, such as TiDB Lightning. Some TiDB tools are released separately from the TiDB server and use their own version numbering system, such as TiUP and TiDB Operator.
```
Comment (lilin90) [inline releases/versioning.md:62]:
```suggestion
Some TiDB tools are released together with the TiDB server and use the same version numbering system, such as TiDB Lightning. Some TiDB tools are released separately from the TiDB server and use their own version numbering system, such as TiUP and TiDB Operator.
```

File: releases/versioning.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+TiDB versioning has the form of `X.Y.Z`. `X.Y` represents a release series.
+
+- Since TiDB 1.0, `X` increments every year. Each `X` release introduces new features and improvements.
+- `Y` increments from 0. Each `Y` release introduces new features and improvements.
+- In the first release of a release series, `Z` is set to 0 by default. For patch releases, `Z` increments from 1.
+
+For the versioning system of TiDB v5.0.0 and earlier versions, refer to [Historical versioning](#historical-versioning-deprecated).
+
+## Long-Term Support Releases
```
Comment (lilin90) [inline releases/versioning.md:27]:
```suggestion
## Long-Term Support releases
```

File: releases/versioning.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+Long-Term Support Releases (LTS) are released approximately every six months and introduce new features, improvements, bug fixes and security vulnerability fixes.
```
Comment (lilin90) [inline releases/versioning.md:29]:
```suggestion
Long-Term Support (LTS) versions are released approximately every six months and introduce new features, improvements, bug fixes and security vulnerability fixes.
```

File: releases/versioning.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+LTS are versioned as `X.Y.Z`. `Z` defaults to 0.
```
Comment (lilin90) [inline releases/versioning.md:31]:
```suggestion
LTS releases are versioned as `X.Y.Z`. `Z` defaults to 0.
```

File: releases/versioning.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+<Note>
+v5.1.0, v5.2.0, v5.3.0, v5.4.0 were released only two months after their preceding releases, but all four releases are LTS and provide patch releases.
+</Note>
+
+## Development Milestone Releases
+
+Development Milestone Releases (DMR) are released approximately every two months that do not contain LTS. DMR introduce new features, improvements and bug fixes. TiDB does not provide patch releases based on DMR, and any related bugs are fixed in the subsequent release series.
+
+DMR are versioned as `X.Y.Z`. `Z` defaults to 0. A `-DMR` suffix is appended to the version number.
```
Comment (lilin90) [inline releases/versioning.md:54]:
```suggestion
DMR releases are versioned as `X.Y.Z`. `Z` defaults to 0. A `-DMR` suffix is appended to the version number.
```

File: releases/versioning.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+- 6.1.1
+
+<Note>
+v5.1.0, v5.2.0, v5.3.0, v5.4.0 were released only two months after their preceding releases, but all four releases are LTS and provide patch releases.
+</Note>
+
+## Development Milestone Releases
+
+Development Milestone Releases (DMR) are released approximately every two months that do not contain LTS. DMR introduce new features, improvements and bug fixes. TiDB does not provide patch releases based on DMR, and any related bugs are fixed in the subsequent release series.
```
Comment (lilin90) [inline releases/versioning.md:52]:
```suggestion
Development Milestone Releases (DMR) are released approximately every two months that do not contain LTS. DMR versions introduce new features, improvements and bug fixes. TiDB does not provide patch releases based on DMR, and any related bugs are fixed in the subsequent release series.
```

File: releases/versioning.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+- 6.0.0-DMR
+
+## Versioning of TiDB ecosystem tools
+
+Some TiDB tools are released together with TiDB server and use the same version numbering system, such as TiDB Lightning. Some TiDB tools are released separately from the TiDB server and use their own version numbering system, such as TiUP and TiDB Operator.
+
+## Historical versioning (deprecated)
+
+### General Availability Releases
```
Comment (lilin90) [inline releases/versioning.md:66]:
```suggestion
### General Availability releases
```

File: releases/versioning.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+
+General Availability Releases (GA) are stable versions of the current release series of TiDB. GA are released after Release Candidate Releases (RC). GA can be used in production environments.
```
Comment (lilin90) [inline releases/versioning.md:68]:
```suggestion
General Availability (GA) releases are stable versions of the current release series of TiDB. GA versions are released after Release Candidate (RC) versions. GA can be used in production environments.
```

File: releases/versioning.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+
+Example versions:
+
+- 1.0
+- 2.1 GA
+- 5.0 GA
+
+### Release Candidate Releases
```
Comment (lilin90) [inline releases/versioning.md:76]:
```suggestion
### Release Candidate releases
```

File: releases/versioning.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+
+Release Candidate Releases (RC) introduce new features and improvements. RC are significantly more stable than Beta releases. RC can be used for early testing, but are not suitable for production.
```
Comment (lilin90) [inline releases/versioning.md:78]:
```suggestion
Release Candidate (RC) releases introduce new features and improvements. RC versions are significantly more stable than Beta versions. RC can be used for early testing, but are not suitable for production.
```

File: releases/versioning.md [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+
+Example versions:
+
+- RC1
+- 2.0-RC1
+- 3.0.0-rc.1
+
+### Beta Releases
```
Comment (lilin90) [inline releases/versioning.md:86]:
```suggestion
### Beta releases
```

File: releases/versioning.md [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+
+Beta Releases introduces new features and improvements. Beta Releases are greatly improved over Alpha Releases and have eliminated critical bugs, but still contain some bugs. Beta Releases are available for users to test the latest features.
+
+Example versions:
+
+- 1.1 Beta
+- 2.1 Beta
+- 4.0.0-beta.1
+
+### Alpha Releases
```
Comment (lilin90) [inline releases/versioning.md:96]:
```suggestion
### Alpha releases
```

File: releases/versioning.md [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+
+Alpha Releases are internal releases for testing and introduce new features and improvements. Alpha releases are the initial versions of the current release series. Alpha Releases might have some bugs and are available for users to test the latest features.
+
+Example versions:
```
Comment (lilin90) [inline releases/versioning.md:100]:
```suggestion
Alpha releases are internal releases for testing and introduce new features and improvements. Alpha releases are the initial versions of the current release series. Alpha releases might have some bugs and are available for users to test the latest features.

Example version:
```

**Review by lilin90 — APPROVED** [review]
LGTM

**Pull request merged**